src/Features/Core/Portable/AddRequiredParentheses/AddRequiredParenthesesCodeFixProvider.cs

```diff
@@ -29,9 +29,9 @@ public override ImmutableArray<string> FixableDiagnosticIds
 
         internal sealed override CodeFixCategory CodeFixCategory => CodeFixCategory.CodeStyle;
 
-        protected override bool IncludeDiagnosticDuringFixAll(FixAllState state, Diagnostic diagnostic, CancellationToken cancellationToken)
```

Comment:
note: this split was intentional to help make it totally clear that this cancellation token was what you shoudl use in this operation and you shouldn't accidentally capture it.

the problem with FixAllContext was that some people ended up capturing it and using the wrong cancellation token later.  i think we should undo this change (or pull it into a separate PR where we can assess it).

Reply:
I can attempt to revert it, but it seemed much cleaner to have all public types used in this virtual method signature, which is defined on the common code fix provider sub-type. FixAllState was supposed to be implementation detail for the code fix engine, not something to be handed to individual code fixes.

Reply:
Sort of.  I was the one that broke them apart back here: https://github.com/dotnet/roslyn/pull/10780

The reason was as above.  FixAllContext was very unsafe and we def had code capture it and use its cancellation token inappropriately.  I'd rather us not revert to that. :-/

Reply:
Alright, so I pushed another commit to take a common route - neither FixAllState nor FixAllContext are now in the signature. Instead we pass in only the publically available pieces from context object into that API.